### ISSUE TITLE
Script checks are not triggering Tasks to run

### ISSUE THREAD

**Issue by technoicon** [wh1te909/tacticalrmm#668]
**Server Info (please complete the following information):**
 - OS: Ubuntu 18.04
 - Browser: Chrome
 - RMM Version (as shown in top left of web UI): 0.7.2

**Installation Method:**
  - [ ] Docker

**Agent Info (please complete the following information):**
- Agent version: Agent v1.5.9
- Agent OS: Windows 7 Professional, 32 bit (build 7601.24546)

**Describe the bug**
I have a few checks setup that trigger tasks if they fail.  A few examples are:
Power shell version < 2 - Triggers powershell upgrade script.
Traylauncher installed - Triggers tray launcher install script.
AV installed check - Triggers AV installation.
I have checked a few machines for one client and it seems that the checks are failing but the tasks are not triggering.  I can run them Manually and they work.

**To Reproduce**
1. Create a policy.
2. Create a check
3. create a task that executes on the check failure.

**Expected behavior**
Expect the tasks to trigger or error.

**Screenshots**

**Additional context**

**Comment by dinger1986** [comment]
Has this been tested on supported os's? IE Ubuntu 20.04 and windows 10?

**Comment by silversword411** [comment]
What powershell <2 script are you using? If it's reliable, mind sharing so I can get it in the community script library?

**Comment by technoicon** [comment]
yeah sure
check powershell Version:
```
if ($PSVersionTable.PSVersion.Major -gt 2)
{
    $PSVersionTable.PSVersion.Major
    Write-Output "PSVersion Greater than 2.0"
    exit 0
}
else
{
    $PSVersionTable.PSVersion.Major
    Write-Output "PSVersion less than 2.0"
    exit 1
}
```
Then if fails < 2.0
```

#save this file and self-host https://download.microsoft.com/download/6/F/5/6F5FF66C-6775-42B0-86C4-47D41F2DA187/Win7AndW2K8R2-KB3191566-x64.zip

if ($PSVersionTable.PSVersion.Major -lt 5)
{
    Write-Output "Old Version - Need to Upgrade"
    # Download MSU file - EDIT THIS URL
   # $url = "www.mustuploadtoyourownsite.com"
    (new-object System.Net.WebClient).DownloadFile($url,'C:\temp\Win7AndW2K8R2-KB3191566-x64.msu')

    ## Run upgrade process
    start-process -FilePath "c:\windows\system32\wusa.exe" -ArgumentList "c:\temp\Win7AndW2K8R2-KB3191566-x64.msu /quiet /norestart /log:c:\temp\log.evt"
    Write-Output "Run upgrade process"
}
else
{
    Write-Output "Already at 5.0 or Higher"
}
```

**Comment by technoicon** [comment]
As far as i'm aware ubuntu 18.04 is supported, that's why I'm using it.

Yes, same problem with windows 10 machines.

see attached.
![check failed](https://user-images.githubusercontent.com/17463110/131346819-3a8cb9e6-faff-423e-aa3a-d2e1a1a9784c.png)
![task never triggered](https://user-images.githubusercontent.com/17463110/131346825-7043036f-7581-43e7-a774-f21f135deb14.png)
![winversion](https://user-images.githubusercontent.com/17463110/131346827-e594c101-946e-4b5d-a8c6-30338986698d.png)

**Comment by silversword411** [comment]
> yeah sure

Ok, that looks familiar. Already got it in the wip...but since it requires manual hosting of files it's not in official location. 

I did add a bunch of other URL's to the commends for other versions. Probably could be enhanced to detect OS, and then download directly from Microsoft instead :)

**Comment by wh1te909** [comment]
confirmed bug, am able to reproduce. for now solution is to create the check/task on the agent level, not with automation manager, until we get this fixed.

**Comment by wh1te909** [comment]
fixed in https://github.com/wh1te909/tacticalrmm/commit/181891757e88082ded5cbab9cf389570c9f9cee8
tasks with schedule "run on check failure" were not being created at all when applied via a policy, so on next release you'll need to recreate them

**Comment by wh1te909** [comment]
fixed in release 0.8.3
